### ISSUE TITLE
make import-sql command does not progress past a certain point

### ISSUE THREAD

**Issue by biskwikman** [openmaptiles/openmaptiles#1272]
When I try to create tiles for Japan, I seem to get stuck whenever I get to the `import-sql` step. I've tried manually doing each step, as well as using `./quickstart.sh japan`.

It gets to a point where it will have something like this:

```
CREATE INDEX
Time: 7.204 ms
CREATE INDEX
Time: 4.651 ms
CREATE VIEW
Time: 2.229 ms
CREATE FUNCTION
Time: 2.300 ms
psql:/sql/parallel/building.sql:314: NOTICE:  Finished layer building
DO
Time: 0.149 ms
```
and will not progress. I've left it for about 10 hours and there was no change. I've been able to successfully create tiles for albania so I don't quite know what the issue is. I'm on Ubuntu 20.04.

**Comment by Falke-Design** [comment]
hmm the log looks good.
Maybe @nyurik, @TomPohys or @ZeLonewolf has an idea what the issue can be

**Comment by nyurik** [comment]
We could always create a magical script that runs as a separate process and automatically queries postgres server for the status/usage, and prints something every 10 minutes with "server CPU/MEM/list of running queries" status... PRs welcome :)

**Comment by TomPohys** [comment]
Hi, from the log there is `Finished` statement for these layers 
```
DONE: aerodrome_label
DONE: aeroway
DONE: boundary
DONE: housenumber
DONE: landuse
DONE: park
DONE: transportation
DONE: water

building
landcover
mountain_peak
place
poi
transportation_name
water_name
waterway
```
The `building` layer could take a long time. Please try comment out building layer in [openmaptiles.yaml](https://github.com/openmaptiles/openmaptiles/blob/master/openmaptiles.yaml#L12) then run `make clean` and try to run `make import-sql` again. You can also comment out the layers which are already done from the previous attempts.

with `MAX_PARALLEL_PSQL=1` the log should be more straightforward to read because it tries to import one SQL at time.

**Comment by nanshanli** [comment]
I've noticed the same thing myself while trying to generate tiles for Japan. After about 75h hours, it was still stuck on importing the `poi` layer.

I noticed the osm.pbf extracts were available for [sub-regions](http://download.geofabrik.de/asia/japan.html) in Japan, but `asia/japan/chubu` does not import after 11-12 hours too.

I am now trying to comment out the `poi` layer to generate the tiles without that layer.

The generation is running on an Ubuntu 20.04 `c3.8xlarge` instance (16 vCPU, 60GB RAM).

**Comment by xinyuluo** [comment]
@nanshanli any updates after removing the `poi` layer? did it complete? I am having a similar issue now. I am left with three layers: `place`, `poi`, and `waterway`. I'm not sure if I got "stuck" or these layers just take really long time to finish.

> the only way to see what's going on is to connect to postgres via a separate terminal, and get all currently running queries

@nyurik I tried to log in postgres in a separate terminal and ran `select * from pg_stat_activity`, didn't find any meaningful queries. is this the correct way to check?

**Comment by biskwikman** [comment]
@xinyuluo sorry I forgot to report back but I tried removing the poi layer and a bunch of other ones (bulidings, waterways) and it would still get stuck.

**Comment by nanshanli** [comment]
@xinyuluo hi - yes, `import-sql` for japan did complete for me after removing the `poi` layer. I think the whole process took 1d 5h in total according to the logs, which is a lot longer than usual.

**Comment by biskwikman** [comment]
@nanshanli oh that's interesting. I'll have to try to let mine run for a few days to see what happens.

**Comment by ZeLonewolf** [comment]
I'm curious if the ubuntu 20.04 used here is a bare-metal install of ubuntu or virtualized in Windows Subsystem for Linux (WSL).  I've found my bare-metal ubuntu runs of OMT to be very fast, while the WSL version is sometimes fast and sometimes slow for no apparent reason.

**Comment by xinyuluo** [comment]
@ZeLonewolf could you provide a benchmark for "very fast" please? for instance, what region of map data did you process, and roughly how long it takes? thank you.

**Comment by nanshanli** [comment]
@ZeLonewolf yes, it was a bare metal instance spun up on AWS. I have generated tiles for all other regions in the world and this problem is the most prominent when generating tiles for Japan. China is also relatively slow at > half a day, but Japan’s import went for 3 days without completing before I removed the poi layer.

**Comment by ZeLonewolf** [comment]
> could you provide a benchmark for "very fast" please? for instance, what region of map data did you process, and roughly how long it takes?

I just ran a test using California as the region.  With a pre-downloaded pbf, I ran `make clean destroy-db && time ./quickstart california` using default openmaptiles.yaml and .env settings (zoom 7, all layers).  Total time was 23m 50s.

**Comment by TomPohys** [comment]
Hi, I just finished some testing and the Chubu area (350MB pbf) was generating very slow. The layers `waterway`, `place`, and `poi` took 3.5h, 1.5, and 1.5h, respectively. After some digging, it seems that the `update_tags` function in these layers took all the time.

https://github.com/openmaptiles/openmaptiles/blob/d7ea45349e6ba05f415d78baedb25596f17010d5/layers/waterway/update_waterway_linestring.sql#L8

https://github.com/openmaptiles/openmaptiles/blob/d7ea45349e6ba05f415d78baedb25596f17010d5/layers/place/update_city_point.sql#L50

https://github.com/openmaptiles/openmaptiles/blob/d7ea45349e6ba05f415d78baedb25596f17010d5/layers/poi/update_poi_point.sql#L26

**Comment by ZeLonewolf** [comment]
I'm not sure what those update functions do exactly, but they sure look complicated!  It doesn't surprise me that they take a long time to run.  I am surprised that the performance is especially poor in one specific area of Japan that doesn't show up in other similarly sized areas, like California.

**Comment by TomPohys** [comment]
This is duplicity for https://github.com/openmaptiles/openmaptiles/issues/967. There is also a discussion and suggested workaround.

**Comment by maxammann** [comment]
I'm not sure if my issue is related but I can not get past the "buliding" layer:
```
Time: 9.333 ms
psql:/sql/parallel/building.sql:314: NOTICE:  Finished layer building
DO
Time: 0.285 ms
 update_osm_park_polygon
-------------------------

(1 row)

Time: 21741362.040 ms (06:02:21.362)
CREATE INDEX
```

I'm importing europe and the import is already running for 14 days now. The VM is not that strong on CPU, but has 4 cores and 16GB or RAM. Swap is currently not used.

Strangely postgres is only utilizing one core, which in fact could not normal. It is currently creating a table.

**Comment by slutske22** [comment]
I am having the same issue with the `boundary` layer:

```
psql:/sql/parallel/boundary.sql:444: NOTICE:  materialized view "ne_50m_admin_0_boundary_lines_land_gen_z2" does not exist, skipping
DROP MATERIALIZED VIEW
Time: 0.362 ms
SELECT 387
Time: 10.612 ms
CREATE INDEX
Time: 5.780 ms
psql:/sql/parallel/boundary.sql:458: NOTICE:  materialized view "ne_50m_admin_0_boundary_lines_land_gen_z1" does not exist, skipping
DROP MATERIALIZED VIEW
Time: 0.347 ms
SELECT 387
Time: 13.323 ms
CREATE INDEX
Time: 5.861 ms
psql:/sql/parallel/boundary.sql:473: NOTICE:  materialized view "ne_110m_admin_0_boundary_lines_land_gen_z0" does not exist, skipping
DROP MATERIALIZED VIEW
Time: 0.327 ms
SELECT 330
Time: 12.058 ms
CREATE INDEX
Time: 4.432 ms
CREATE FUNCTION
Time: 7.368 ms
CREATE VIEW
Time: 4.009 ms
CREATE VIEW
Time: 5.569 ms
CREATE VIEW
Time: 3.969 ms
CREATE VIEW
Time: 3.751 ms
CREATE VIEW
Time: 4.771 ms
CREATE VIEW
Time: 3.165 ms
CREATE VIEW
Time: 3.103 ms
CREATE VIEW
Time: 2.907 ms
CREATE VIEW
Time: 2.985 ms
CREATE VIEW
Time: 2.958 ms
CREATE VIEW
Time: 2.769 ms
CREATE VIEW
Time: 2.682 ms
CREATE VIEW
Time: 3.678 ms
CREATE VIEW
Time: 4.193 ms
CREATE FUNCTION
Time: 12.049 ms
psql:/sql/parallel/boundary.sql:898: NOTICE:  Finished layer boundary
DO
Time: 0.965 ms
```

I'm on an M2 mac 96 GB RAM.  I started a `./quickstart planet.sh` for z0-12 on monday morning, was running smooth for about 3 days, then got stuck here.

The shame of this is that there doesn't seem to be any output, so to try again means to start from the beginning?

@TomPohys you suggested that #967 is the same issue, but I'm not too clear on what the workaround there is?  Is it to run off of https://github.com/osm-by/openmaptiles/commit/5e229e6794d9dbd839a3eeedbf6a2d41c224b077, and rerun the quickstart with that commit?
